### ISSUE TITLE
MenuToggle - add aria-invalid and aria-autocomplete attributes to examples

### ISSUE THREAD

**Issue by nikkimk** [patternfly/patternfly#6562]
Task list (only targeting v6):

- [ ] Add `aria-invalid="true"` to any MenuToggle showing a danger status (to the button element of the toggle). If a typeahead toggle is showing danger status, this should be applied to the input element.
- [ ] Add `aria-autocomplete="list"` to the text input of a typeahead menu toggle

Original comment below:

**Describe the problem**
This is an issue with 
- [x] Patternfly 5  
- [ ] Patternfly 6  
- [x] other (Patternfly 4)

Based on a review by @adamjohnson and me, we have determined that dropdown has a number of accessibility issues

**How do you reproduce the problem?**
***Single select***
1. Go to the [single select example](https://www.patternfly.org/components/menus/select#single-select).
2. Note that the `role` is missing, making the purpose of the element unclear to screenreader users who cannot see the screen.
3. Note that `aria-controls` is missing.

***With validation***
1. Go to the [validation example](https://www.patternfly.org/components/menus/select#with-validation).
2. Select an option.
3. Note that for screenreader users who cannot see the screen, the select is not semantically associated with the validation message.

***Checkbox select***
1. Go to the [validation example](https://www.patternfly.org/components/menus/select#checkbox-select).
2. Note that the `role` is missing, making the purpose of the element unclear to screenreader users who cannot see the screen.
3. Note that `aria-controls` is missing.
4. Note that the popup element, while functioning identically to a multiselectable listbox, is not implemented that way semantically in a way that is familiar to screenreader users.
5. Note that the disabled option is not focusable but should be.

**Expected behavior**
***All variants of select***
- should have a [`role="combobox"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role)
- should have a label that describes the combobox. (**Note:** that the selected option or a placeholder option **_should not_** be used as a label)
- should have an [`aria-controls`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) set to the IDREF of the element that pops up
- should use [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) to indicate whether the popup is displayed
- should use [`aria-activedescendant`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-activedescendant) with the IDREF of the selected option **or** a [roving tabindex](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#Keyboard%20Navigation%20Inside%20Components) with the selected option as the active option
- should use a listbox in the popup element (See WAI ARIA Authoring Practices Guide's [https://www.w3.org/WAI/ARIA/apg/patterns/listbox/].)
- options in a listbox should still be focusable when disabled (See WAI ARIA Authoring Practices Guide's [Focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols).)

***Single select***
- See the WAI ARIA Authoring Practices Guide's [select-only combobox example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).

***Checkbox select***
- should use [`aria-multiselectable="true"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-multiselectable) 
- the popup element should be a listbox (See WAI ARIA Authoring Practices Guide's [https://www.w3.org/WAI/ARIA/apg/patterns/listbox/].)
- the checkboxes (which serve as visual feedback of which options are selected) should be `aria-hidden` and the menuitems themselves should be focusable/selectable

***Validation***
- for _invalid_ selections, should have [`aria-invalid="true"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-invalid)
- for _invalid_ selections, should have [`aria-errormessage`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage) with an IDREF to validation message
- for _validated_ selections, should have [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) with an IDREF to validation message

***Typeahead***
- should use [`aria-autocomplete`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-autocomplete) for typeahead
- element should have a label that is distinct from placeholder and selected value
- See WAI ARIA Authoring Practice Guide's [combobox with authocomplete](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/).

**Is this issue blocking you?**
No

**Any other information?**
This creates accessibility issues wherever implemented.

cc/ @markcaron

**Comment by thatblindgeye** [comment]
Any aria-controls and aria-haspopup issues should be covered by https://github.com/patternfly/patternfly/issues/6561 and associated React followups.

Comments for each issue/expectation are below, based off v6 as we do not plan to make updates for v4 and any updates for v5 would most likely only be made if it's a breaking bug for consumers.

Single select:
- Role is present so not an issue

With validation:
- This is still an issue, and should be fixed as part of https://github.com/patternfly/patternfly-react/issues/11470

Checkbox select:
- Role is present so not an issue
- This implementation is a little different than a typical select. The roles should be valid as the purpose is to utilize checkboxes as the means of conveying selection, rather than a typical `listbox` and `options` roles. Perhaps worth revisiting whether it makes sense to have this as a variant of Select or not, but a separate and more in-depth convo.
- isDisabled should be used for fully disabling, whereas isAriaDisabled should be used if the item should remain focusable. This shouldn't be considered an issue.

Expecations:
- I don't think we should add the combobox role to any MenuToggle being used for a Select. VoiceOver announces "type text or, to display a list of options..." whether the role is on an input or a button, which is confusing. While the ARIA example does use this role, their example has logic where typing when the toggle has focus opens it and jumps to the first option that matches that character; our Select/MenuToggle does not have that functionality and would probably rely on the consumer to implement since their 2 separate composable components. The aria-haspopup attribute should suffice in conveying the purpose of the toggle.
- Because of the above item, `aria-activedescendant` doesn't need to be added anywhere else as our Typeaheads should already include it in example code (the attribute per MDN should only be on combobox, group, application, and textbox roles)

Based on the above + how the Select components currently works in v6 and intentions of it, I think the expected changes needed to be made are as follows:

- Update MenuToggle docs to 1) mention that if the MenuToggle content is expected or intended to dynamically update, there needs to be a static label describing the purpose of the toggle, and 2) update examples/demos accordingly (probably just Select component, since Dropdown typically has a static toggle text)
- Add an `aria-invalid="true"` to MenuToggle when the status is danger. `aria-describedby` should be handled by https://github.com/patternfly/patternfly-react/issues/11470 (seems like aria-errormessage doesn't have great support, and I don't think we should use that instead of something more widely supported like aria-describedby; aria-invalid + the describedby should be sufficient)
- Add `aria-autocomplete="list"` to the input elements of a Typeahead Select
